### ISSUE TITLE
Hiding multiple schematics at once via template manager

### ISSUE THREAD

**Issue by Swanky1337** [X2CommunityCore/X2WOTCCommunityHighlander#1161]
I've been trying to set up a system within Iri's Template Manager that hides schematics for items in lower tiers once a schematic for higher tiers is purchased.
However, it would always only hide the item directly benath the purchased one (or the specified item), not cascading down on any other open schematics that may have come before. So if there are Tier 1.5, 2, 2.5 and Tier 3 items open because they have never been purchased, buying the T3 item would only hide the specified T2.5 item, but not all that came before.

 Reference:
+Edit_X2ItemTemplate = (T = "AssaultRifle_T2_Schematic", P = "HideIfPurchased", V = "AssaultRifle_T3_Schematic")
;+Edit_X2ItemTemplate = (T = "AssaultRifle_T2_Schematic", P = "HideIfPurchased", V = "AssaultRifle_T4_Schematic")
;+Edit_X2ItemTemplate = (T = "AssaultRifle_T2_Schematic", P = "HideIfPurchased", V = "AssaultRifle_T5_Schematic")
+Edit_X2ItemTemplate = (T = "AssaultRifle_T3_Schematic", P = "HideIfPurchased", V = "AssaultRifle_T4_Schematic")
;+Edit_X2ItemTemplate = (T = "AssaultRifle_T3_Schematic", P = "HideIfPurchased", V = "AssaultRifle_T5_Schematic")
+Edit_X2ItemTemplate = (T = "AssaultRifle_T4_Schematic", P = "HideIfPurchased", V = "AssaultRifle_T5_Schematic")

Usually, this isn't a big issue but with added item tiers it can generally clog up the menus on the Avenger (and potentially elsewhere).
Speaking to Zelfana on discord, they identified the following as a potential point to expand and fix although they weren't sure as to how.

function bool MeetsDisplayRequirement(X2ItemTemplate ItemTemplate)
{
    local XComGameState_HeadquartersXCom XComHQ;

    XComHQ = class'UIUtilities_Strategy'.static.GetXComHQ();

    return (!XComHQ.IsTechResearched(ItemTemplate.HideIfResearched) && !XComHQ.HasItemByName(ItemTemplate.HideIfPurchased));
}

Message log:
https://discord.com/channels/165245941664710656/402042994255069184/1068221338705219627

**Comment by Iridar** [comment]
Yeah, this is because `HideIfResearched` / `HideIfPurchased` are not arrays, so you can't include several schematics in there, and the logic doesn't do a cascade check. The game kinda assumes you won't be skipping squad upgrades.

The best thing that can be done in Highlander about it is to put an override event into `MeetsDisplayRequirement()` so that other mods can use arbitrary logic to hide items, including a cascade check.
